drivers/pinctrl/pinctrl_imx.c

```diff
@@ -79,6 +149,8 @@ int pinctrl_configure_pins(const pinctrl_soc_pin_t *pins, uint8_t pin_cnt,
 	return 0;
 }
 
+#endif /* CONFIG_ARM_SCMI_PINCTRL_HELPERS */
+
 static int imx_pinctrl_init(void)
```

Comment:
bracket the calling of HAL clock helpers by the condition !CONFIG_ARM_SCMI_PINCTRL_HELPERS.
But I prefer to create a new driver for SCMI-based one and leave this driver as it's.

Reply:
For me, either keeping one driver or adding a new driver is okay.
I will add a new driver to address your above comments.
Thanks.